unity/Holo/Assets/ModelClippingPlaneControl.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using HoloToolkit.Unity.InputModule;
+using HoloToolkit.Unity.UX;
+
+public class ModelClippingPlaneControl : MonoBehaviour, IClickHandler
+{
+    public GameObject ClippingPlane;
+    public GameObject ClippingPlaneModifyBtn;
+
+    public GameObject LoadedModel;
+
+    BoundingBoxRig clipPlaneQuadBbox;
+    Material currentModelMaterial;
+    Material currentModelMaterialClip;
+    
+    bool clipPlaneOn = false;
+    bool clipPlaneModifyActive = false;
+
+    void Start()
+    {
+        clipPlaneQuadBbox = ClippingPlane.GetComponent<BoundingBoxRig>();
+        ClippingPlane.GetComponent<HandDraggable>().enabled = false;
+
+        ResetState();
+    }
+
+    public void Click(GameObject clickObj)
+    {
+        Debug.Log("Clicked obj: " + clickObj.name);
+        switch (clickObj.name)
+        {
+            case "ButtonClipping":
+                ClickedClipPlane();
+                break;
+
+            case "ButtonClippingManipulator":
+                ClickedClipPlaneModify();
+                break;
+        }
+
+    }
+
+    void ClickedClipPlaneModify()
+    {
+        Debug.Log("Toggle modification of clip plane");
+        clipPlaneModifyActive = !clipPlaneModifyActive;
+        if (clipPlaneModifyActive)
+        {
+            clipPlaneQuadBbox.Activate();
+            ClippingPlane.GetComponent<HandDraggable>().enabled = true;
+        }
+        else
+        {
+            clipPlaneQuadBbox.Deactivate();
+            ClippingPlane.GetComponent<HandDraggable>().enabled = false;
+        }
+    }
+
+    void ClickedClipPlane()
+    { 
+        const string materialsPath = "Materials/";
+        GameObject currentModel = GameObject.Find("mainModel");
+        if (!currentModel)
+        {
+            Debug.Log("No model loaded for clipping plane");
+            clipPlaneOn = false;
+            return;
+        }
+
+        SkinnedMeshRenderer renderer = currentModel.GetComponent<SkinnedMeshRenderer>();
+        if (!clipPlaneOn)
+        {
+            currentModelMaterial = renderer.material;
+            string clipMat = currentModelMaterial.name + "Clip";
+            clipMat = clipMat.Replace(" (Instance)", ""); // We instantiate models from prefabs.
+            clipMat = materialsPath + clipMat;

```

Comment:
Possibly this determination of material name can be simplified.

Right now we use just 2 materials for models (`MaterialPreview` and `MaterialNonPreview`, public references in ModelWithPlate.cs). And this is only for testing, to differentiate between preview and non-preview, in the future I think we will have just 1 material (when clipping is off), i.e. `MaterialPreview` will be removed. The "preview" model will just have a simpler mesh, as that is the purpose of "preview" model version.

So, I think you can just use here a single, always the same, material. It can be exposed and assigned like MaterialPreview in ModelWithPlate.cs , just `public Material MaterialClip` in this script and in editor you can drag-and-drop proper material asset (with the shader using clipping).

This will be more complicated once we have more meshes (for simulations etc.). But I propose for start let's merge to master clipping plane on one mesh working cool, and improve from there :)